component/useProductList.js

```diff
@@ -1,31 +1,39 @@
-import { useEffect, useState } from 'react';
+import {useEffect, useState, useRef} from 'react';
 
 const useProductList = () => {
   const [products, setProducts] = useState([]);
   const [loading, setLoading] = useState(true);
   const [skip, setSkip] = useState(0);
+  const [isFetchingMore, setIsFetchingMore] = useState(false);
 
   const fetchProducts = async () => {
     try {
-      const res = await fetch(`https://dummyjson.com/products?skip=${skip}`);
+      setLoading(true);
+      const res = await fetch(
+        `https://dummyjson.com/products?limit=10&skip=${skip}`,
+      );
       const data = await res.json();
-      setProducts((prevProducts) => [...prevProducts, ...data?.products]);
+      setProducts(prevProducts => [...prevProducts, ...data?.products || []]);
     } catch (error) {
       console.error('Error fetching products:', error);
     } finally {
       setLoading(false);
+      setIsFetchingMore(false);
     }
   };
 
   const loadMore = () => {
-    setSkip((prevSkip) => prevSkip + 5); 
+    if (!isFetchingMore) {
+      setIsFetchingMore(true);
+      setSkip(prevSkip => prevSkip + 5);
+    }
```

Comment:
why is skip is equal to 5 when we have limit of 10 ?

Reply:
Done @mohity777